app-foundation/4-projects/development/boa_secret.tf

```diff
@@ -0,0 +1,135 @@
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+module "boa_secret_project" {
+  source                      = "github.com/terraform-google-modules/terraform-example-foundation/4-projects/modules/single_project"
+  impersonate_service_account = var.terraform_service_account
+  org_id                      = var.org_id
+  billing_account             = var.billing_account
+  folder_id                   = data.google_active_folder.env.name
+  environment                 = "development"
+  vpc_type                    = "base"
+  enable_hub_and_spoke        = var.enable_hub_and_spoke
+  alert_spent_percents        = var.alert_spent_percents
+  alert_pubsub_topic          = var.alert_pubsub_topic
+  budget_amount               = var.budget_amount
+  project_prefix              = var.project_prefix
+  activate_apis = [
+    "cloudresourcemanager.googleapis.com",
+    "monitoring.googleapis.com",
+    "logging.googleapis.com",
+    "cloudkms.googleapis.com",
+    "secretmanager.googleapis.com",
+    "compute.googleapis.com",
+    "container.googleapis.com",
+    "cloudbilling.googleapis.com",
+    "iam.googleapis.com",
+    "serviceusage.googleapis.com",
+    "logging.googleapis.com",
+    "storage-api.googleapis.com",
+    "sqladmin.googleapis.com",
+    "servicenetworking.googleapis.com"
+  ]
+
+  # Metadata
+  project_suffix    = "boa-sec"
+  application_name  = "bu1-sample-application"
+  billing_code      = "1234"
+  primary_contact   = "example@example.com"
+  secondary_contact = "example2@example.com"
+  business_code     = "bu1"
+}
+
+module "terraform_deployment_sa" {
+  source     = "terraform-google-modules/service-accounts/google"
+  version    = "~> 3.0"
+  project_id = module.boa_secret_project.project_id
+  names      = ["boa-terraform-${var.environment_code}-sa"]
+  project_roles = [
+    "${module.boa_gke_project.project_id}=>roles/compute.viewer",
```

Comment:
Can we create a locals block and group these by projects to make it more readable? Something like (syntax maybe off I havent tested)

```tf
locals {
tf_deploy_sa_roles = {
boa_gke_project_roles = [...roles/compute.viewer],
boa_secret_project = [...roles]
}
}
project_roles = [for project, roles in local.tf_deploy_sa_roles : [for role in roles: "${project}=>${role}"]
```